File: power_of_thor/taille_min/pot_taille.m
#import <Foundation/Foundation.h>
int main(){
  int lx,ly,tx,ty;
  scanf("%d%d%d%d",&lx,&ly,&tx,&ty);
  while (1){
    printf("%s%s\n",(ty<ly?"S":ty>ly?"N":""),(tx<lx?"E":tx>lx?"W":""));
    tx+=(tx<lx)-(tx>lx);
    ty+=(ty<ly)-(ty>ly);
    scanf("%*d");}
}

File: power_of_thor/taille_min/pot_taille.c
#include <stdio.h>
int main(){
 int lx, ly, tx, ty, t;
 scanf("%d %d %d %d",&lx,&ly,&tx,&ty);
 while(scanf("%d",&t)!=EOF){
  printf("%s%s\n",(ly>ty)-(ly<ty)>0?"S":(ly>ty)-(ly<ty)<0?"N":"",(lx>tx)-(lx<tx)>0?"E":(lx>tx)-(lx<tx)<0?"W":"");
  tx+=(lx>tx)-(lx<tx);
  ty+=(ly>ty)-(ly<ty);}
 return 0;
}
